### PR TITLE
TAR-71: replaced deprecated call to GG profile.

### PR DESCRIPTION
**Pull request by paoloboni**
GET /profile endpoint is now deprecated. This call has been replaced with:

- call to auth - to get user enrolments
- call to user-details - to get affinity group

Urls are provided by the authority object.

### CODE REVIEW

File: app/model/CoAFEAuthority.scala [hunk 1]
```diff
@@ -16,10 +16,15 @@
...
 
-case class CoAFEAuthority(twoFactorAuthOtpId: Option[String])
+case class CoAFEAuthority(twoFactorAuthOtpId: Option[String], enrolmentsUri: String = "", userDetailsLink: String = "")
 
 object CoAFEAuthority {
-  implicit val reads = Json.reads[CoAFEAuthority]
+  implicit val reads =
+    ((JsPath \ "twoFactorAuthOtpId").readNullable[String] and
+      (JsPath \ "enrolments").read[String] and
+      (JsPath \ "userDetailsLink").read[String]).apply(CoAFEAuthority.apply _)
```
Comment (anshulbajpai) [inline app/model/CoAFEAuthority.scala]: Wouldn't it read as expected automatically without this logic?

File: app/model/CoAFEAuthority.scala [hunk 2]
```diff
@@ -16,10 +16,15 @@
 
 package model
 
+import play.api.libs.functional.syntax._
+import play.api.libs.json.Reads._
 import play.api.libs.json._
 
-case class CoAFEAuthority(twoFactorAuthOtpId: Option[String])
+case class CoAFEAuthority(twoFactorAuthOtpId: Option[String], enrolmentsUri: String = "", userDetailsLink: String = "")
```
Comment (anshulbajpai) [inline app/model/CoAFEAuthority.scala]: Why are their default values?

File: it/router/RouterAuditFeature.scala [hunk 3]
```diff
@@ -476,12 +484,50 @@ class RouterAuditFeature extends StubbedFeatureSpec with CommonStubs {
...
     }
+
+    scenario("when a user logged in through GG and has no sa and no business enrolment and affinity group n/a an audit event should be raised") {
+
+      Given("a user logged in through Government Gateway")
+      val saUtr = "12345"
+      val accounts = Accounts(sa = Some(SaAccount("", SaUtr(saUtr))))
+      createStubs(TaxAccountUser(accounts = accounts, affinityGroup = AffinityGroupValue.INDIVIDUAL))
+
+      And("the user has self assessment enrolments and individual affinity group")
```
Comment (anshulbajpai) [inline it/router/RouterAuditFeature.scala]: Description does not match the intent

File: it/router/RouterAuditFeature.scala [hunk 4]
```diff
@@ -476,12 +484,50 @@ class RouterAuditFeature extends StubbedFeatureSpec with CommonStubs {
...
+        HAS_ANY_INACTIVE_ENROLMENT.key -> "false",
+        AFFINITY_GROUP_AVAILABLE.key -> "true"
         ))
 
       val expectedTransactionName = "sent to personal tax account"
       verifyAuditEvent(auditEventStub, expectedReasons, expectedTransactionName, "pta-home-page-individual")
     }
+
+    scenario("when a user logged in through GG and has no sa and no business enrolment and affinity group n/a an audit event should be raised") {
+
```
Comment (anshulbajpai) [inline it/router/RouterAuditFeature.scala]: n/a usually means "not applicable"

File: app/model/CoAFEAuthority.scala (hunk 1 above)
Comment (paoloboni) [inline app/model/CoAFEAuthority.scala]: Custom deserialiser is needed here as `enrolments` field is translated into `enrolmentsUri`.

**Pull request merged**